### PR TITLE
restructure jenkins core/plugin installs along api.ci vs. osbs/brew l…

### PR DESCRIPTION
**Pull request by gabemontero**
…ines

See https://github.com/openshift/ocp-build-data/pull/48

/assign @bparees 
@openshift/sig-developer-experience fyi
@adammhaile @sosiouxme fyi

### CODE REVIEW

**Comment by gabemontero** [comment]
/hold

this should merge after https://github.com/openshift/ocp-build-data/pull/48

File: 2/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM quay.io/openshift/origin-cli
+FROM openshift/origin:v3.11
+#FROM quay.io/openshift/origin-cli
```
Comment (bparees) [inline 2/Dockerfile]: why is this changing?  (also referencing v3.11 seems odd)

File: 2/contrib/jenkins/install-plugins.sh [hunk 2]
```diff
@@ -1,436 +1,8 @@
...
-
-    if [[ -f $FAILED ]]; then
-        echo -e "\nSome plugins failed to download!\n$(<"$FAILED")" >&2
-        exit 1
-    fi
-
-    echo -e "\nCleaning up locks"
-    rm -rf "$REF_DIR"/*.lock
+    mainInstallPlugins "$@"
 }
```
Comment (bparees) [inline 2/contrib/jenkins/install-plugins.sh]: doesn't seem like you need to define main() at all.  Just call mainInstallPlugins below.

File: 2/Dockerfile (hunk 1 above)
Comment (gabemontero) [reply 2/Dockerfile]: ah ... that was suppose to be temporary as I didn't want to wrestle with the UBI stuff as well ... will revert

File: 2/contrib/jenkins/install-plugins.sh (hunk 2 above)
Comment (gabemontero) [reply 2/contrib/jenkins/install-plugins.sh]: Yeah I wasn't sure re: bash nuances if that would somehow mess up existing s2i extensions that leverate install-plugins.sh 

Assuming you don't think of any in the next few minutes, I'll make this change

**Comment by gabemontero** [comment]
Getting rid of `main()` might address this as welll, but what happened when from `install-jenkins-core-plugins.sh`
- sourced install-plugins.sh
- called `install-jenkins-core-plugins.sh` from the dockerfile
it would bypass my logic in `install-jenkins-core.sh` and call the `main()` in `install-plugins.sh`

if, when getting rid of `main()` in `install-plugins.sh`, the all the logic in `install-jenkins-core-plugins.sh` is called, I'll converge back to just `install-plugins.sh`

**Comment by bparees** [comment]
This seems to do what i'd expect and i think roughly maps to what you're trying to accomplish:


```
$ cat wrapper.sh 
#!/bin/sh

source ./inner.sh

echo wrapperlogic
# invokes main from inner.sh
main
```

```
$ cat inner.sh 
#!/bin/sh

main() {
  echo innermain
}

# this gets executed when inner.sh is run, or when inner.sh is sourced
echo inner not function scoped
```

```
$ ./wrapper.sh 
inner not function scoped
wrapperlogic
innermain
```

``
$ ./inner.sh 
inner not function scoped
```

**Comment by bparees** [comment]
so i think the issue you hit is that when you source the "inner" script, it contains an invocation of main itself.

File: 2/contrib/jenkins/install-jenkins-core-plugins.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#! /bin/bash -eu
+
+source /usr/local/bin/install-plugins-internals.sh
```
Comment (bparees) [inline 2/contrib/jenkins/install-jenkins-core-plugins.sh]: part of the solution to your issue might be to not source this unless you're in the rpms=false case.

Comment (gabemontero) [reply 2/contrib/jenkins/install-jenkins-core-plugins.sh]: I was about to suggest that ... we still want folks to be able to call inner.sh directly .. i.e. install-plugins.sh directly ... as they may have been doing that in their extension scenarios

Comment (gabemontero) [reply 2/contrib/jenkins/install-jenkins-core-plugins.sh]: assuming I can source within the logic like that and pass the parameters

Comment (bparees) [reply 2/contrib/jenkins/install-jenkins-core-plugins.sh]: https://stackoverflow.com/questions/39768712/how-to-pass-arguments-to-a-script-invoked-by-source-command

Comment (bparees) [reply 2/contrib/jenkins/install-jenkins-core-plugins.sh]: or you can just call the script rather than source it.

Comment (gabemontero) [reply 2/contrib/jenkins/install-jenkins-core-plugins.sh]: the latter is working ... will push that shortly

**Comment by gabemontero** [comment]
update pushed @bparees ptal

**Comment by gabemontero** [comment]
@adammhaile the jenkins PR for our ocp-build-data replace has merged ... please re-add it back

**Pull request merged**